### PR TITLE
Rename toJsonObject() to toJsonValue(), fromJsonObject() to fromJsonV…

### PR DESCRIPTION
**Pull request by swankjesse**
…alue().

Also rename the JsonReader and JsonWriter implementations.

### CODE REVIEW

File: moshi/src/main/java/com/squareup/moshi/JsonUt8Writer.java [hunk 1]
```diff
@@ -27,7 +27,7 @@
 import static com.squareup.moshi.JsonScope.NONEMPTY_DOCUMENT;
 import static com.squareup.moshi.JsonScope.NONEMPTY_OBJECT;
 
-final class BufferedSinkJsonWriter extends JsonWriter {
+final class JsonUt8Writer extends JsonWriter {
```
Comment (stevecstian) [inline moshi/src/main/java/com/squareup/moshi/JsonUt8Writer.java]: You meant `JsonUtf8Writer`, right?

File: moshi/src/test/java/com/squareup/moshi/JsonUt8WriterTest.java [hunk 2]
```diff
@@ -21,7 +21,7 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 
-public final class BufferedSinkJsonWriterTest {
+public final class JsonUt8WriterTest {
```
Comment (stevecstian) [inline moshi/src/test/java/com/squareup/moshi/JsonUt8WriterTest.java]: You meant `JsonUtf8WriterTest`, right?

File: moshi/src/main/java/com/squareup/moshi/JsonUt8Writer.java (hunk 1 above)
Comment (swankjesse) [inline moshi/src/main/java/com/squareup/moshi/JsonUt8Writer.java]: YEP. Ouch.

Comment (swankjesse) [inline moshi/src/main/java/com/squareup/moshi/JsonUt8Writer.java]: https://github.com/square/moshi/pull/249

**Pull request merged**